tests/unit/test_xblock_poll.py

```diff
@@ -140,3 +178,69 @@ def test_student_view_user_state_handler(self):
             },
         }
         self.assertEqual(response, expected_response)
+
+    @classmethod
+    def mock_user_states(cls):
+        return (
+            Mock(
+                username='edx',
+                state={
+                    'submissions_count': 1,
+                    'choices': {'enjoy': 'Y', 'recommend': 'N', 'learn': 'M'}
+                }
+            ),
+            Mock(
+                username='verified',
+                state={
+                    'submissions_count': 1,
+                    'choices': {'enjoy': 'M', 'recommend': 'N', 'learn': 'Y'}
+                }
+            ),
+            Mock(
+                username='staff',
+                state={
+                    'submissions_count': 1,
+                    'choices': {'enjoy': 'N', 'recommend': 'N', 'learn': 'N'}
+                }
+            ),
+            Mock(
+                username='honor',
+                state={
+                    'submissions_count': 1,
+                    'choices': {'enjoy': 'Y', 'recommend': 'N', 'learn': 'M'}
+                }
+            ),
+        )
+
+    def test_generate_report_data_dont_limit_responses(self):
+        """
+        Test generate_report_data iterator with no limit.
+        """
+        user_states = self.mock_user_states()
+        report_data = self.survey_block.generate_report_data(user_states)
+        report_data = list(report_data)
+        self.assertEqual(len(report_data), 12)
+        # each choice of a user gets its own row
+        # so the first three rows should be edx's choices
+        self.assertEqual(['edx', 'edx', 'edx'],
+                         [username for username, _ in report_data[:3]])
```

Comment:
Could you bump this up to 4 so we get responses for 2 users?

Reply:
Updated to assert 2 users